### ISSUE TITLE
Minimize data sent in HTTP user agent

### ISSUE THREAD

**Issue by hecktest** [minetest/minetest#14830]
Split from #14819 

### Problem

The client makes HTTP requests to various built-in services (master server, content DB) to fetch data.
The requests use a custom user agent which includes the exact OS version, which is completely unnecessary to the operation of the services, since MT uses a platform-independent scripting runtime and network protocol. This data cannot be used for any legitimate purposes while creating unnecessary liability, such as being personally identifiable.

### Solutions

Reduce the user-agent string to just "minetest" with a numbered version. This version is the only useful metric that's actually worth collecting and measuring. Ensure that this sanitized user-agent is used for every HTTP request.

### Alternatives

-

### Additional context

What is the user agent used for HTTP asset fetches, if a server exposes those? We might want to especially sanitize this, because a malicious/compromised server could use this to select a payload.

**Comment by SingleDigitIQ** [comment]
Just my two cents,
> This version is the only useful metric that's actually worth collecting and measuring. Ensure that this sanitized user-agent is used for every HTTP request.

It needs a hash then, an id, so that several devices on one address will be identifiable without exposing actual "user-agent". Also, even for those who agrees to telemetry, ideally requests for updates should be done once per day maximum.

Also it's 2024 already, learn to double-think. I think that asking for an update only once per day would be going too far, tracking is necessary. You will see: as Minetest grows in popularity(if it happens at all), eventually it will arrive at the point of no return, and core devs will be politely asked by glowing people to add tracking to make sure safety of players, probably children in particular. You see there is a special issue with Minetest from glowing person perspective: absolutely anybody can run a server. Not implementing HTTPS is sort of a disclaimer in this case.
Maybe now it's all nice and reversible, maybe not and tracking will continue until morale improves. Sometimes devs are being asked impolitely if polite requests do not work, I remember some people had them hacked mysteriously in a very spooky way, because they were too much into free software and speech. They are docile now, or at least seem so.

I think current tracking system is perfect. You seem to assume that criminals are probably rational by default or something, while in fact most are irrational, they do not try to hide from tracking or anything like that, they autopilot something stupid like bots or glowing people would do. Against that kind of crime, so the majority of all crime, this tracking can help. Core devs can provide data to authorities, stay compliant and authority-friendly.

Just needs consent, and it would be very nice to have huge obfuscated "terms of service" something for people to never read and agree on everything :>

**Comment by sfan5** [comment]
> This data cannot be used for any legitimate purposes

That depends on your definition but I think producing version and platform statistic data is very much "legitimate".

What this has helped with in the past:
* abuse prevention in some cases (and no, the attacker did not try spoofing it)
* checking how well new MT versions reach our users
* deciding when to drop older network protocols depending on actual usage
* deciding how old of OS versions we need to support

I can envision the following:
* since windows is now more or less rolling release build numbers can become important for whether a given feature is available
* if we decide to add arch-specific optimizations (think AVX2), statistics could decide what to focus on

> such as being personally identifiable.

As far as I can tell the user agent is not PII as defined by GDPR ([ref](https://law.stackexchange.com/questions/54929/is-the-user-agent-string-considered-personal-data)).
It might become a problem when combined with a logged IP address, though that on its own seems to be fine long as you do it for security purposes and do not otherwise process them ([ref](https://law.stackexchange.com/questions/28603/how-to-satisfy-gdprs-consent-requirement-for-ip-logging)).

Still, I agree with minimizing this data so that a malicious actor could do less with it.

> What is the user agent used for HTTP asset fetches, if a server exposes those? 

The same as the others.

<hr>

My suggestion:
* nerf the MT version number so it just says "x.x.x-dev" or "x.x.x". Git hashes are useless and too unique.
* cut off kernel version on Unix-like to pure numbers (`6.9.7-arch1-1` -> `6.9.7`)
* make Android send `Android/<api level>` instead of the kernel version. This is both more useful to know and less unique.

I am open to getting rid of the windows build number, though it could have future use.

**Comment by rubenwardy** [comment]
I don't really care about information given to the update detection API, so I would be fine with only sending Minetest version in the request there. Especially if it would increase adoption of the update detection feature

Platform information when loading the ContentDB dialog and installing packages is more useful and more interesting to me. Android comes with a different set of default filters to hide mature content not allowed by Google Play.

At some point, I'd like to be able to tell content creators the proportion of desktop vs mobile installs on the stats page

**Comment by sfan5** [comment]
> I don't see a legitimate scenario in which this occurs; do we really want to help Microsoft make ReactOS a perpetually moving target?

Honestly yes: this is a very minor theoretical scenario I came up with.

Especially now that we're on SDL I don't forsee us directly interacting with win32 in a way that new W10-only features will become relevant.
Though it's worth noting that there were things like `inet_pton` not being available on WinXP that are now behind us since the minimum is W7.

FWIW: I don't think Microsoft has an agenda to keep ReactOS behind, though it might suit them. It's quite natural that applications target the versions of the real thing (Windows) that are actually in use and that emulators have to keep up with it to be actually useful.

>  I'm fine with a mobile/nonmobile bit, it could even be exposed to game servers. 

The client exposes its touchscreen status to the server right now: #14092
